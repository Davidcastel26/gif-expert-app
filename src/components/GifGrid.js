import React /*, {useState, useEffect}*/ from 'react'
import PropTypes from 'prop-types';

import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';

const GifGrid = ({category}) => {

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'> {category} </h3>

            { loading && <p className='animate__animated animate__flash'>Loading at this moment bro...</p>}

            <div className='card-grid'>
                {
                    images.map( img =>(
                        <GifGridItem 
                        key={ img.id}
                        {...img}
                        /> 
                    ))
                }
            </div>
        </>
    )
}

 GifGrid.propTypes = {
     category: PropTypes.string.isRequired,
 }

export default GifGrid;









// import getGifs from '../helpers/getGifs'

    // const [count, setCount] = useState(1)
    // console.log(data, loading);

/*const [images, setImages] = useState([]) */


// useEffect( () => {
    // getGifs( category )
        // .then(imgs => setImages(imgs))
        // .then( setImages )
// }, [category])


// images.map( ({id, title}) =>(
    // <li key={ id}> { title} </li>
// ))

// {/* <h3> { count } </h3> */}
// {/* <button onClick={ () => setCount( count + 1)}></button> */}