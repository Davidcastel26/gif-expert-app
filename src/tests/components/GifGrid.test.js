// react imports 
import React from 'react';
import GifGrid from '../../components/GifGrid';
//testing imports
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';


describe('testing the Gif Grid component', () => {

    const category = 'jest.fn';
    let wrapper = shallow( < GifGrid category={category}/> )

    test('should show the right path', ()=>{
        expect( wrapper ).toMatchSnapshot();
    })

})