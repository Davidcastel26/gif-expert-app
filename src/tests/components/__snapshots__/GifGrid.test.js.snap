// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Gif Grid component should show the right path 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    jest.fn
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading at this moment bro...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
